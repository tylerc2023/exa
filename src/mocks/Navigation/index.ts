export const mockNavigation = {
  goBack: jest.fn(),
  navigate: jest.fn(),
  push: jest.fn(),
  popToTop: jest.fn(),
};
