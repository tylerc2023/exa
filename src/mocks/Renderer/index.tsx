import React from 'react';
import {Provider} from 'react-redux';
import {ReactTestRenderer, create} from 'react-test-renderer';
import {store} from '../../providers/Redux';

export const mockRenderer = (component: JSX.Element): ReactTestRenderer =>
  create(<Provider store={store}>{component}</Provider>);
