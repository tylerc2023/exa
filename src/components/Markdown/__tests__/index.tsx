import React from 'react';
import {Markdown} from '..';
import {mockRenderer} from '../../../mocks/Renderer';

it('renders correctly', () => {
  const {tree} = mockRenderer({
    component: <Markdown title="hello *world*" />,
  });
  expect(tree.toJSON()).toMatchSnapshot();
});
