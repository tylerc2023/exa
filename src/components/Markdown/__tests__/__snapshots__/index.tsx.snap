// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Text
    style={
      Object {
        "color": "hsla(0, 0%, 0%, 1)",
        "fontSize": 14,
        "fontWeight": "300",
        "letterSpacing": 0.25,
        "opacity": 1,
      }
    }
  >
    hello 
  </Text>
  <Text
    style={
      Object {
        "color": "hsla(0, 0%, 0%, 1)",
        "fontSize": 14,
        "fontWeight": "600",
        "letterSpacing": 0.25,
        "opacity": 1,
      }
    }
  >
    world
  </Text>
</View>
`;
